### DISCUSSION TITLE
Error when setting schema composite ID with a relation

### DISCUSSION THREAD

**Discussion by ashemag** [prisma/prisma#12328] [Q&A]
Hey there, 

I'm having trouble setting a composite id (@@unique([userId, email]) when userId is a relation. The issue arrises when I try to delete a user with this setup. It throws an error, I'm guessing because the cascade is causing a problem with this field. 

Anyone have any references or thoughts on this? Any help appreciated! Details below  

Schema: 
```prisma
model Contact {
  userId          String    @map("user_id")
  createdAt       DateTime  @default(now())
  displayName     String?
  firstName       String?
  lastName        String?
  email           String
  photoUrl        String?

  user            User      @relation(fields: [userId], references: [id], onDelete: Cascade)
  @@unique([userId, email], name:"id")
  @@map("contacts")
}
```

Error: 
```
Type: undefined
Message: 
Invalid `prisma.user.delete()` invocation:


  Error occurred during query execution:
ConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(Server(ServerError { code: 1105, message: "symbol `hearth-db`.users.id not found", state: "HY000" })) })

Code: undefined

Query:
[object Object]
```

<img width="1477" alt="Screen Shot 2022-03-14 at 2 23 16 PM" src="https://user-images.githubusercontent.com/8096590/158236633-bf6b0c67-ebec-43de-838e-296333923b91.png">

**Comment by austincrim** [comment]
Hey @ashemag 👋,

I was unable to recreate this issue locally. What database provider are you using?

Also, if you are using the combo of `userId` and `email` as an ID, you might want to read about our [`@@id` attribute.](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#id-1) as well.

Thanks!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
